rate_counter/rate/views.py

```diff
@@ -1,3 +1,159 @@
-from django.shortcuts import render
+from django.shortcuts import render, HttpResponseRedirect, get_object_or_404, get_list_or_404
+from django.core.urlresolvers import reverse
+from .forms import TableForm
+from .models import Institute, Subject, StudentGroup, Table, TableRow
+from .utils import get_result, get_color
+
 
 # Create your views here.
+
+
+def create_table(request):
+    """
+    creation of new table
+    """
+    form = TableForm(request.POST or None)
+    if form.is_valid():
+        instance = form.save(commit=False)
+        instance.save()
+        return HttpResponseRedirect(reverse('rate:table_detail', args=[str(instance.id)]))
+    context = {
+        'form': form
+    }
+    return render(request=request, template_name="table_form.html", context=context)
+
+
+def edit_table(request, id=None):
+    """
+    edit of existing table
+    """
+    table_obj = get_object_or_404(Table, id=id)
+    form = TableForm(request.POST or None, request.FILES or None, instance=table_obj)
+    if form.is_valid():
+        instance = form.save(commit=False)
+        instance.save()
+        return HttpResponseRedirect(reverse('rate:table_detail', args=[str(instance.id)]))
+    context = {
+        'form': form
+    }
+    return render(request=request, template_name="table_form.html", context=context)
+
+
+def delete_table(request, id=None):
+    """
```

Comment:
@Scandie now it is totally ok to use function based views, because they are very small. In a future we might start use [Django class based views](https://docs.djangoproject.com/en/1.9/ref/class-based-views/generic-editing/#django.views.generic.edit.DeleteView). No need to do it now, just keep in mind.